File: PumpLogUI/src/models/exercise.ts
export interface BodyPart {
  bodyPartGuid: string;
  name: string;
}

export interface Exercise {
  exerciseGuid: string;
  exerciseName: string;
  name: string;
  description?: string;
  bodyPartGuid: string;
  bodyPart?: BodyPart;
}

File: PumpLogApi/Entities/Exercise.cs
using System.ComponentModel.DataAnnotations;

namespace PumpLogApi.Entities
{
    public class Exercise
    {
        [Key]
        public Guid ExerciseGuid { get; set; }
        public required string Name { get; set; }
        public string? Description { get; set; }
        public Guid BodyPartGuid { get; set; }
        public BodyPart? BodyPart { get; set; }
    }
}
